openpype/plugins/publish/integrate_new.py

```diff
@@ -110,801 +156,494 @@ class IntegrateAssetNew(pyblish.api.InstancePlugin):
+            "outputName": "output",
+        }.items():
+            value = repre.get(representation_key)
+            if value:
+                template_data[anatomy_key] = value
+
+        if repre.get('stagingDir'):
+            stagingdir = repre['stagingDir']
+        else:
+            # Fall back to instance staging dir if not explicitly
+            # set for representation in the instance
+            self.log.debug("Representation uses instance staging dir: "
+                           "{}".format(instance_stagingdir))
+            stagingdir = instance_stagingdir
+        if not stagingdir:
+            raise ValueError("No staging directory set for representation: "
+                             "{}".format(repre))
 
-            Uses Settings > Global > Publish plugins > IntegrateAssetNew
+        self.log.debug("Anatomy template name: {}".format(template_name))
+        anatomy = instance.context.data['anatomy']
+        template = os.path.normpath(anatomy.templates[template_name]["path"])
 
-            Args:
-                instance (dict): processed instance
-                subset_id (str): DB's subset _id
+        is_sequence_representation = isinstance(files, (list, tuple))
+        if is_sequence_representation:
+            # Collection of files (sequence)
+            assert not any(os.path.isabs(fname) for fname in files), (
+                "Given file names contain full paths"
+            )
 
-        """
-        # Fist look into instance data
-        subset_group = instance.data.get("subsetGroup")
-        if not subset_group:
-            subset_group = self._get_subset_group(instance)
+            src_collection = assemble(files)
 
-        if subset_group:
-            io.update_many({
-                'type': 'subset',
-                '_id': ObjectId(subset_id)
-            }, {'$set': {'data.subsetGroup': subset_group}})
+            # If the representation has `frameStart` set it renumbers the
+            # frame indices of the published collection. It will start from
+            # that `frameStart` index instead. Thus if that frame start
+            # differs from the collection we want to shift the destination
+            # frame indices from the source collection.
+            destination_indexes = list(src_collection.indexes)
+            destination_padding = len(get_first_frame_padded(src_collection))
+            if repre.get("frameStart") is not None:
+                index_frame_start = int(repre.get("frameStart"))
 
-    def _get_subset_group(self, instance):
-        """Look into subset group profiles set by settings.
+                # TODO use frame padding from right template group
+                render_template = anatomy.templates["render"]
+                frame_start_padding = int(
+                    render_template.get(
+                        "frame_padding",
+                        render_template.get("padding")
+                    )
+                )
```

Comment:
Similar to before the refactor the Render Template padding is only used when the `frameStart` value is defined in the representation data in the instance. Is this to be expected? Or would we expect the Render Template Frame Padding to **always** be used for sequences even when `frameStart` is not set?

Reply:
Can you ever get a sequence without `frameStart`?